uktrade/data-hub-frontend
# Add EYB lead collection list

## Description of change

This PR adds a collection list to the EYB lead tab page. It also enables users to filter EYB leads by company name, top level sector, and value of the lead (high or low).

The API work to enable the filtering can be found in:
-  https://github.com/uktrade/data-hub-api/pull/5661
- https://github.com/uktrade/data-hub-api/pull/5670

The bulk of the diff (approx 600 lines) is from fixture data and updating the sandbox API schema, which is meant to be used in the lead faker.

## Test instructions

1. Populate the database with some EYB leads (use the EYBLeadFactory in the API)
2. Navigate to `Investments -> EYB leads`
3. You should see a list of the EYB leads
4. You should be able to filter the list by any combination of company name, sector, and value
5. You should be able to go to a individual lead's detail page by clicking the lead title (i.e. company name its associated with).

### Screenshot

<img width="797" alt="image" src="https://github.com/user-attachments/assets/94fa8ada-38eb-4e51-b799-52ba78637528">

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
